[manifest] Added manifest for deploying on aws using s3

eterna2 opened the pull request:
There are a number of requests both in the issues (#2627 #1610 #1131) as well as in the slack channel on how to configure kubeflow pipeline on AWS (namely interacting with S3 buckets, particularly for pipeline-ui).

This PR adds 2 kustomize overlays:
- accesskey: overlay that reference a secret with aws credentials
- iam: overlay that annotate the services with the iam role (assuming kube2iam or similar are already deployed in the cluster)

The more common configurations (e.g. bucket, folder) are also exposed inside `params.env`. `archiveLogs` for argo is also set to `true` so that the logs are persisted and can be retrieved by the ui.

A README is also provided in the root dir of the overlays.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/pipelines/2633)
<!-- Reviewable:end -->

Ark-kun commented on manifests/kustomize/env/aws/accesskey/aws-configurations-patch.yaml:
```diff
@@ -0,0 +1,93 @@
+      containers:
+        - name: ml-pipeline-ui
+          env:
+            - name: AWS_ACCESS_KEY_ID
```
@IronPan In future we should probably switch to getting the keys from ConfigMap.

Bobgy replied:
access_key_id and access_key sounds like they belong to a secret. I think this is appropriate.

Ark-kun replied:
> access_key_id and access_key sounds like they belong to a secret.

AFAIK, they're not the secret itself, more like secret name.

P.S. If the Frontend currently gets those configurations from the ENV, then I guess this PR just reflects that. We should move them to configMap when we have a chance.

Ark-kun commented on manifests/kustomize/env/aws/accesskey/aws-configurations-patch.yaml:
```diff
@@ -0,0 +1,93 @@
+                  key: $(awsSecretKeySecretKey)
+            - name: ARGO_ARCHIVE_LOGS
+              value: "true"
+            - name: ARGO_ARCHIVE_ARTIFACTORY
```
@Bobgy @IronPan  I wonder why UI is not getting these values from ConfigMap. Probably the configmap did not exist at the time.

Bobgy commented on manifests/kustomize/env/aws/accesskey/viewer-tensorboard-template-configmap.yaml:
```diff
@@ -0,0 +1,38 @@
+                        },
+                        {
+                            "name": "AWS_REGION",
+                            "value": "ap-southeast-1"
```
Should this be configurable?

eterna2 replied:
Yeah. Missed this.

eterna2 commented:
Good idea. I will do that and revert this commit and update the readme instead.

Jeffwan commented on manifests/kustomize/env/aws/iam/aws-configurations-patch.yaml:
```diff
@@ -0,0 +1,78 @@
+  template:
+    metadata:
+      annotations:
+        iam.amazonaws.com/role: $(awsIAMRole)
```
Let's make the `key` configurable?  `iam.amazonaws.com/role` is for k2iam users. As EKS release feature https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/, we have lots of users use `eks.amazonaws.com/role-arn` as well.

eterna2 replied:
kustomize don't allow templating for keys yet - i.e. I cannot do `$(key): $(value)` for annotation.

a possible workaround is to use `commonAnnotations` (actually this is what I used for my deployment, cuz I was lazy. I gave all kfp deployments the same iam role)

eterna2 commented:
Hey all, 

Created a new repo: https://github.com/e2fyi/kubeflow-aws to hold the manifest. Streamlined the manifest too.

- most of the configurations are done in `base`
- variant:accesskey just patches the various deployment with the access key secret
- variant:iam only patches the tensorboard configmap, and annotations are added using `commonAnnotations` 

For this PR:
- i reverted the previous commit (remove all the manifest)
- Added folder `env/aws` and a README inside (easier for pple to spot if they are looking at the repo)
- Update the manifest README with a small section under customization

Bobgy commented on manifests/kustomize/README.md:
```diff
@@ -48,7 +48,30 @@ kubectl port-forward -n kubeflow svc/ml-pipeline-ui 8080:80
+provides a community-maintained manifest for deploying kubeflow pipelines on AWS
+(with S3 as artifact store instead of minio).
+
+TL;DR
```
Shall we also move these instructions into the new repo?
I think that makes it more self contained, and instruction + manifests are versioned together makes more sense

Ark-kun replied:
We should probably still have at least a small section about AWS and a pointer to the other repo. @IronPan WDYT?

Bobgy replied:
Agree, I was thinking the section only introduces the repo.

eterna2 commented:
Ok. I have removed the unnecessary deployment instructions in the readme.

Bobgy commented:
/lgtm
/approve

Thanks! this looks good

Bobgy commented:
/ok-to-test
